### DISCUSSION TITLE
how would I add a script tag?

### DISCUSSION THREAD

**Discussion by vital-tech-results** [vercel/next.js#21093] [Help]
How would I add a ```<script>``` tag in nextjs? 

I am using a ```create react app``` theme that has this in the ```head``` of the ```index.html```

```
<script>
        if (localStorage.theme === 'dark' || !('theme' in localStorage)) {
            document.querySelector('html').classList.add('dark');
        } else {
            document.querySelector('html').classList.remove('dark');
        }        
    </script>
```

I tried adding this in a ```<Head>``` element in my ```_document.js``` file but I got an error; something about ```unexpected token``` regarding the semicolon after ```('dark');```

Thank you,

**Comment by vital-tech-results** [comment]
I think this answers my question:

https://github.com/vercel/next.js/blob/canary/examples/with-google-analytics/pages/_document.js

**Reply by vassbence** [reply]
Yes, exactly, you can even use [next/head](https://nextjs.org/docs/api-reference/next/head)

Like:

```js
<Head>
  <script
    dangerouslySetInnerHTML={{
      __html: `
          if (localStorage.theme === 'dark' || !('theme' in localStorage)) {
              document.querySelector('html').classList.add('dark');
          } else {
              document.querySelector('html').classList.remove('dark');
          }   
    `,
    }}
  />
</Head>
```
